### ISSUE TITLE
Unprivileged Update File System

### ISSUE THREAD

**Issue by mpass99** [openHPI/poseidon#226]
- Test files need to be readable by `user`, as this user is supposed to execute them. Hence, we cannot remove that flag.
- With regard to the example you provided: Yes, the current implementation is a simplification. However, I am currently not aware of a use case where appending is required, but this might be valid. Instead of adding an explicit owner, group and permissions, I would go a soft way: What if we add one additional boolean flag (besides `path` and `content`) to the UpdateFileRoute indicating whether the user used for execution should have write permissions? Or which permissions (`rwx`) the user should haven general? I would just prevent providing an explicit user name, as this requires further configuration. If set, this could be apply to the "other" group and will therefore work for the `user`. If privilegedExecution is enabled, the "other" group won't be respected, but `root` may always read :)

_Originally posted by @MrSerth in https://github.com/openHPI/poseidon/pull/215#discussion_r982869770_

**Comment by MrSerth** [comment]
Indeed, I was not aware of that (but remember the issue; that's one of the reasons the Java image used to call `javac` as `root` (back in the old days with DockerContainerPool). 

I would see this issue as another one as the bug; this issue is about the _file_ permissions, whereas the bug is about the _folder_ permissions, isn't it? Can we solve the folder permissions by either granting more permissions (i.e., +w for others) or changing the owner for these new folders?

Something we should also consider: Allowing write permissions in a folder for anyone could render the ownership for files useless: Even though files might not be writable directly, one could delete and recreate them. 

Regardless of that and a solution, I would prioritize the bug to resolve the issue with our production environment.

**Comment by MrSerth** [comment]
I created #236 for the bug.

**Comment by MrSerth** [comment]
Remember: This issue is about the _file_ permissions copied by Poseidon. So far, we always copy files as `root`, which means that `user` cannot touch these. However, when an exercise requires an **existing** file to be extended (such as a log file or CSV, ...), this might cause trouble. Therefore, we intended to enhance the copy files route with a third attribute:

```patch
     {
        path: file.filepath,
        content: Base64.strict_encode64(file.read || ''),
+    privileged: true
      }
```
